packages/components/src/components/ViewerMenuBar/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState } from "react";
+
+import { SqValuePath } from "@quri/squiggle-lang";
+
+import { SqOutputResult } from "../../../../squiggle-lang/src/public/types.js";
+import { SquiggleOutput } from "../../lib/hooks/useSquiggle.js";
+import { CodeEditorHandle } from "../CodeEditor/index.js";
+import { PartialPlaygroundSettings } from "../PlaygroundSettings.js";
+import { SquiggleViewerHandle } from "../SquiggleViewer/ViewerProvider.js";
+import { Layout } from "./Layout.js";
+import { RenderingIndicator } from "./RenderingIndicator.js";
+import { ViewerBody } from "./ViewerBody.js";
+import { ViewerMenu } from "./ViewerMenu.js";
+
+type Props = {
+  squiggleOutput: SquiggleOutput;
+  isRunning: boolean;
+  editor?: CodeEditorHandle;
+  playgroundSettings: PartialPlaygroundSettings;
+  viewerRef?: React.ForwardedRef<SquiggleViewerHandle>;
```

Comment:
Huh. I assume I accidentally suggested this on call? I thought we were talking about _editor_ ref.

I'm surprised this works at all!

I tried to google for a few minutes to see if maybe `forwardRef` is only useful for native DOM refs. For now, I'm still confused:
- I know that refs are basically objects with `current` props that can be used as variables, so there's a decent chance it's allowed
- but I also know that `ref` involves some magic; https://github.com/reactjs/rfcs/blob/createlement-rfc/text/0000-create-element-changes.md notes that `forwardRef` will be removed over time, but the migration is not over yet, and I believe won't be over even with React 19
- ChatGPT says that it's possible, but it might be hallucinating
- https://stackoverflow.com/questions/68960584/useimperativehandle-usage-for-children-componenent-but-cannot-get-function-from reports a weird problem when `useImperativeHandle` is used without forwarding a ref

Because of that last issue, I think you should `forwardRef`, just in case.

Reply:
Would we need to ``forwardRef`` in both ViewerMenuBar/index and ViewerMenuBar/ViewerBody? If so, maybe just keep it now if it seems to work?

Reply:
Unfortunately, yes, `forwardRef` is necessary on all intermediate components. I know is annoying to write, especially with TypeScript...